### PR TITLE
feat(kit): new `Date` mask

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

related #54

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
+import {DemoPath} from 'projects/demo/src/app/app.routes';
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts:1]:
```suggestion
import {DemoPath} from '@demo/routes';
```

File: projects/demo-integrations/cypress/tests/kit/time/time-basic.cy.ts [hunk 2]
```diff
@@ -222,9 +221,9 @@ describe('Time', () => {
 
                     cy.get('@input')
                         .type('{del}')
-                        .should('have.value', '23:00')
-                        .should('have.prop', 'selectionStart', '23:00'.length)
-                        .should('have.prop', 'selectionEnd', '23:00'.length);
+                        .should('have.value', '23')
+                        .should('have.prop', 'selectionStart', '23'.length)
+                        .should('have.prop', 'selectionEnd', '23'.length);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/time/time-basic.cy.ts:226]: Change test description
```
'23:|59| => Delete => 23:|00'
```
=>
```
'23:|59| => Delete => 23|'
```

File: projects/demo/src/pages/kit/date/date-mask-doc.template.html [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                        Default value is
+                        <code>.</code>
+                        (Dot)
+                    </p>
+                </li>
+            </ul>
+        </section>
+
+        <tui-doc-example
+            id="basic"
```
Comment (nsbarsukov) [inline projects/demo/src/pages/kit/date/date-mask-doc.template.html:61]:
```suggestion
            id="modes"
```

File: projects/kit/src/lib/masks/date/date-mask.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import {MaskitoOptions, maskitoPipe} from '@maskito/core';
+import {MaskitoDateMode} from '@maskito/kit';
```
Comment (nsbarsukov) [inline projects/kit/src/lib/masks/date/date-mask.ts:2]: relative import

File: projects/kit/src/lib/masks/date/processors/max-validation-preprocessor.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import {MaskitoOptions} from '@maskito/core';
+
+import {padWithZeroesUntilValid} from '../../../utils';
+import {DATE_SEGMENT_VALUE_LENGTH} from '../constants';
+import {MaskitoDateSegments} from '../types';
+import {parseDateString} from '../utils';
+import {toDateString} from '../utils/to-date-string';
```
Comment (nsbarsukov) [inline projects/kit/src/lib/masks/date/processors/max-validation-preprocessor.ts:7]: add `toDateString` to `../utils/index.ts`

File: projects/kit/src/lib/masks/time/time-mask.ts [hunk 6]
```diff
@@ -24,7 +22,7 @@ export function maskitoTimeOptionsGenerator({
             TIME_FIXED_CHARACTERS.includes(char) ? char : /\d/,
         ),
         preprocessor: maskitoPipe(
-            createZeroPlaceholdersPreprocessor(mode),
+            createZeroPlaceholdersPreprocessor(),
```
Comment (nsbarsukov) [inline projects/kit/src/lib/masks/time/time-mask.ts:25]: it requires no argument.

I think we can drop redundant JS closure and rename it to:
```suggestion
            zeroPlaceholdersPreprocessor,
```

Comment (nsbarsukov) [reply projects/kit/src/lib/masks/time/time-mask.ts:25]: What do you think ?

Or is it better keeps it as it is (to be similar with other processors)?

File: projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+
+describe('Date', () => {
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/date/date-basic.cy.ts:3]: add tests for these cases, please:

* Empty input => type `9` => `09|`
* `27|` => type `9` => `27.09|`
* `3|` => Type `7` => no value changes

File: projects/demo/src/pages/kit/date/date-mask-doc.template.html [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+            heading="modes"
+        >
+            <tui-input
+                tuiTextfieldCustomContent="tuiIconCalendarLarge"
+                ngModel="02.06.2018"
+                tuiTextfieldFiller="dd.mm.yyyy"
```
Comment (nsbarsukov) [inline projects/demo/src/pages/kit/date/date-mask-doc.template.html:67]: Add fillers for `Time`-page too, please

File: projects/kit/src/lib/masks/time/time-mask.ts (hunk 6 above)
Comment (vladimirpotekhin) [reply projects/kit/src/lib/masks/time/time-mask.ts:25]: I like current varian, let's keep it as it is

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**